### PR TITLE
Add support for using_license_seat field in users

### LINKED ISSUE #1000: Add support for using_license_seat field in users
From GitLab documentation:
Users on GitLab Starter, Bronze, or higher also see the shared_runners_minutes_limit, extra_shared_runners_minutes_limit, and **using_license_seat parameters**.

### PR DESCRIPTION
**Pull request by alex1989hu**
Use type *bool instead of bool because this field is not returned by GitLab API.

This field is only available in GitLab Starter, Bronze, or higher.

Fixes #1000 

Signed-off-by: Alex Szakaly <alex.szakaly@gmail.com>

### CODE REVIEW

File: users.go [hunk 1]
```diff
@@ -89,6 +89,9 @@ type User struct {
 	PrivateProfile            bool               `json:"private_profile"`
 	SharedRunnersMinutesLimit int                `json:"shared_runners_minutes_limit"`
 	CustomAttributes          []*CustomAttribute `json:"custom_attributes"`
+
+	// UsingLicenseSeat is only available in GitLab Starter, Bronze, or higher.
+	UsingLicenseSeat *bool `json:"using_license_seat,omitempty"`
```
Comment (svanharmelen) [inline users.go:94]: Maybe also add `extra_shared_runners_minutes_limit` while you are at it?

But also, please do not make a distinction for certain fields by making them a pointer field. See my comment here #998 as well. So please make it normal fields and place them below `SharedRunnersMinutesLimit` following the order in the docs.

Thanks!

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Next to the inline comment also a small FYI; no need to create an issue first. A descriptive PR is just fine for me 👍🏻

File: users.go (hunk 1 above)
Comment (alex1989hu) [reply users.go:94]: I see your point, to be honest am little bit confused: some codes use pointer field. Sure, I can update this.

Do we want to have `omitempty` at `UsingLicenseSeat` field?

```go
	SharedRunnersMinutesLimit      int                `json:"shared_runners_minutes_limit"`
	ExtraSharedRunnersMinutesLimit int                `json:"extra_shared_runners_minutes_limit"`
	UsingLicenseSeat               bool               `json:"using_license_seat,omitempty"`
```

Comment (svanharmelen) [reply users.go:94]: Nope, no `omitempty` for those please.

As for what should be a pointer and what should not. In options structs everything should be a pointer (so we can use them properly when making API calls) and in response structs only struct fields are pointers. Hope that helps...

Comment (alex1989hu) [reply users.go]: Gotcha, thank you for your explanation. Updated the code as you rested - force pushed the branch.

**Review by svanharmelen — APPROVED** [review]
Thanks! LGTM 👍🏻

**Pull request merged**